applications/nrf5340_audio/src/modules/hw_codec.c

```diff
@@ -316,3 +319,96 @@ int hw_codec_init(void)
 
 	return 0;
 }
+
+static int cmd_input(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t idx;
+
+	enum hw_codec_input {
+		LINE_IN,
+		PDM_MIC,
+		NUM_INPUT,
+	};
+
+	if (argc != 2) {
+		shell_error(shell, "Wrong number of arguments provided %d", argc);
+		return -EINVAL;
+	}
+
+	if (!isdigit((int)argv[1][0])) {
+		shell_error(shell, "Supplied argument is not numeric");
+		return -EINVAL;
+	}
+
+	idx = strtoul(argv[1], NULL, BASE_10);
+
+	if (idx >= NUM_INPUT) {
+		shell_error(shell, "Selected input out of range");
+		return -EINVAL;
+	}
+
+	if ((CONFIG_AUDIO_DEV == GATEWAY) && IS_ENABLED(CONFIG_AUDIO_SOURCE_USB)) {
+		shell_error(shell, "Can't select PDM mic if audio source is USB");
+		return -EINVAL;
+	}
+	if ((CONFIG_AUDIO_DEV == HEADSET) && !IS_ENABLED(CONFIG_STREAM_BIDIRECTIONAL)) {
+		shell_error(shell, "Can't select input if headset is not in bidirectional stream");
+		return -EINVAL;
+	}
+
+	if (idx == LINE_IN) {
+		if (CONFIG_AUDIO_DEV == HEADSET) {
+			ret = cs47l63_comm_reg_conf_write(input_enable, ARRAY_SIZE(input_enable));
+			if (ret) {
+				shell_error(shell, "Failed to enable LINE-IN");
+				return ret;
+			}
+		}
+
+		ret = cs47l63_write_reg(&cs47l63_driver, CS47L63_ASP1TX1_INPUT1, 0x800012);
+		if (ret) {
+			shell_error(shell, "Failed to route LINE-IN to I2S");
+			return ret;
+		}
+
+		ret = cs47l63_write_reg(&cs47l63_driver, CS47L63_ASP1TX2_INPUT1, 0x800013);
+		if (ret) {
+			shell_error(shell, "Failed to route LINE-IN to I2S");
+			return ret;
+		}
+		shell_print(shell, "Selected LINE-IN as input");
+
+	} else if (idx == PDM_MIC) {
+		if (CONFIG_AUDIO_DEV == GATEWAY) {
+			ret = cs47l63_comm_reg_conf_write(pdm_mic_enable_configure,
```

Comment:
Is LINE IN input disabled here?

Reply:
No, but it is not routed to the I2S. I'm not sure how much this would affect the power usage, I don't think it's a lot. If we want to disable the other config we then also have to check if it's already enabled or not etc and if you are already using shell commands we don't care about the power usage since you need USB connected

Reply:
We probably should look at HW codec power usage etc. at some point